server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts

```diff
@@ -678,12 +678,23 @@ export default class ServiceProviderReferralsController {
   }
 
   async createDraftEndOfServiceReport(req: Request, res: Response): Promise<void> {
-    const draftEndOfServiceReport = await this.interventionsService.createDraftEndOfServiceReport(
-      res.locals.user.token.accessToken,
-      req.params.id
-    )
+    const { accessToken } = res.locals.user.token
+    const referralId = req.params.id
+
+    const referral = await this.interventionsService.getSentReferral(accessToken, referralId)
```

Comment:
@tomsmyers do you think it's fine to do this on the Frontend, or should we update the API endpoint to return an existing EOSR if one already exists when we call `createDraftEndOfServiceReport`?

Reply:
we can do this here, but we should also not allow more than one EOSR to be created for the referral in the API. currently it just creates another.